src/circuit/gadget/ecc/chip/mul_fixed/short.rs

```diff
@@ -0,0 +1,251 @@
+            let y_p = meta.query_advice(self.super_config.y_p, Rotation::cur());
+            let y_a = meta.query_advice(self.super_config.add_config.y_qr, Rotation::cur());
+            let sign = meta.query_advice(self.super_config.window, Rotation::cur());
+
+            // `(x_a, y_a)` is the result of `[m]B`, where `m` is the magnitude.
+            // We conditionally negate this result using `y_p = y_a * s`, where `s` is the sign.
+
+            // Check that the final `y_p = y_a` or `y_p = -y_a`
+            let y_check = q_mul_fixed_short.clone()
+                * (y_p.clone() - y_a.clone())
+                * (y_p.clone() + y_a.clone());
+
+            // Check that the correct sign is witnessed s.t. sign * y_p = y_a
+            let negation_check = sign * y_p - y_a;
+
+            array::IntoIter::new([y_check, negation_check])
+                .map(move |poly| q_mul_fixed_short.clone() * poly)
+        });
+    }
+
+    pub fn assign_region(
+        &self,
+        scalar: &EccScalarFixedShort,
+        base: &ValueCommitV,
+        offset: usize,
+        region: &mut Region<'_, pallas::Base>,
+    ) -> Result<EccPoint, Error> {
+        // Copy the scalar decomposition
+        self.super_config
+            .copy_scalar(region, offset, &scalar.into())?;
+
+        let (acc, mul_b) = self.super_config.assign_region_inner(
+            region,
+            offset,
+            &scalar.into(),
+            base.clone().into(),
+            self.super_config.mul_fixed,
+        )?;
+
+        // Add to the cumulative sum to get `[magnitude]B`.
+        let magnitude_mul = self.super_config.add_config.assign_region(
+            &mul_b,
+            &acc,
+            offset + NUM_WINDOWS,
+            region,
+        )?;
+
+        // Increase offset by 1 after complete addition
+        let offset = offset + 1;
+
+        // Assign sign to `window` column
+        let sign = copy(
+            region,
+            || "sign",
+            self.super_config.window,
+            offset + NUM_WINDOWS,
+            &scalar.sign,
+            &self.super_config.perm,
+        )?;
+
+        // Conditionally negate `y`-coordinate
+        let y_val = if let Some(sign) = sign.value() {
+            if sign == -pallas::Base::one() {
+                magnitude_mul.y.value().map(|y: pallas::Base| -y)
+            } else {
+                magnitude_mul.y.value()
+            }
+        } else {
+            None
+        };
+
+        // Enable mul_fixed_short selector on final row
+        self.q_mul_fixed_short
+            .enable(region, offset + NUM_WINDOWS)?;
+
+        // Assign final `x, y` to `x_p, y_p` columns and return final point
+        let x_val = magnitude_mul.x.value();
+        let x_var = region.assign_advice(
+            || "x_var",
+            self.super_config.x_p,
+            offset + NUM_WINDOWS,
+            || x_val.ok_or(Error::SynthesisError),
+        )?;
```

Comment:
Blocking: This cell isn't constrained to equal `magnitude_mul.x`. However, AFAICT we don't actually need to assign it, since we can just return `magnitude_mul.x` (this is the end of the overall region, we don't need the output to be in a specific column).
```suggestion
        // Assign final `y` to `y_p` column, and return final point.
```